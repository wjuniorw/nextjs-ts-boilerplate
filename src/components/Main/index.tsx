function Main() {
  return (
    <main>
      <h1>Nextjs S2...</h1>
    </main>
  )
}

export default Main
