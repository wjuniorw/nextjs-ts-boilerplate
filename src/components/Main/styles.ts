import styled from 'styled-components'

export const Wrapper = styled.main`
  color: #fff;
  width: 100%;
  height: 100%;
  display: flex;
  padding: 3rem;
  text-align: center;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  background-color: #06092b;
`

export const Logo = styled.img`
  width: 70px;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  margin-top: 2rem;
  font-size: 2rem;
  font-weight: 400;
`

export const Ilustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
