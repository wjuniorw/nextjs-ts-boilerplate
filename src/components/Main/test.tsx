import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('should render something...', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Nextjs S2.../i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
