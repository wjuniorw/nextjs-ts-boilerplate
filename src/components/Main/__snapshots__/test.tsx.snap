// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render something... 1`] = `
<main>
  <h1>
    Nextjs S2...
  </h1>
</main>
`;
